One of the CI failures is:

```
F:\workspace\_work\1\s\src\libraries\Common\tests\System\Net\Http\HttpClientHandlerTest.Decompression.cs(49,55): error CS0246:
The type or namespace name 'ZLibStream' could not be found (are you missing a using directive or an assembly reference?) 
[F:\workspace\_work\1\s\src\libraries\System.Net.Http.WinHttpHandler\tests\FunctionalTests\System.Net.Http.WinHttpHandler.Functional.Tests.csproj]


```

--- a/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Decompression.cs
+++ b/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Decompression.cs
@@ -39,42 +39,49 @@ namespace System.Net.Http.Functional.Tests
             }
         }
 
-        public static IEnumerable<object[]> DecompressedResponse_MethodSpecified_DecompressedContentReturned_MemberData()
-        {
-            foreach (bool specifyAllMethods in new[] { false, true })
-            {
-                yield return new object[]
-                {
-                    "deflate",
-                    new Func<Stream, Stream>(s => new ZLibStream(s, CompressionLevel.Optimal, leaveOpen: true)),
-                    specifyAllMethods ? DecompressionMethods.Deflate : _all
-                };
-                yield return new object[]
-                {
-                    "gzip",
-                    new Func<Stream, Stream>(s => new GZipStream(s, CompressionLevel.Optimal, leaveOpen: true)),
-                    specifyAllMethods ? DecompressionMethods.GZip : _all
-                };
-#if !NETFRAMEWORK
-                yield return new object[]
-                {
-                    "br",
-                    new Func<Stream, Stream>(s => new BrotliStream(s, CompressionLevel.Optimal, leaveOpen: true)),
-                    specifyAllMethods ? DecompressionMethods.Brotli : _all
-                };
-#endif
-            }
-        }
-
         [Theory]
-        [MemberData(nameof(DecompressedResponse_MethodSpecified_DecompressedContentReturned_MemberData))]
-        public async Task DecompressedResponse_MethodSpecified_DecompressedContentReturned(
-            string encodingName, Func<Stream, Stream> compress, DecompressionMethods methods)
+        [InlineData("gzip", false)]
+        [InlineData("gzip", true)]
+        [InlineData("deflate", false)]
+        [InlineData("deflate", true)]
+        [InlineData("br", false)]
+        [InlineData("br", true)]
+        public async Task DecompressedResponse_MethodSpecified_DecompressedContentReturned(string encodingName, bool all)
         {
-            // Brotli only supported on SocketsHttpHandler.
-            if (IsWinHttpHandler && encodingName == "br")
+            Func<Stream, Stream> compress;
+            DecompressionMethods methods;
+            switch (encodingName)
             {
-                return;
+                case "gzip":
+                    compress = s => new GZipStream(s, CompressionLevel.Optimal, leaveOpen: true);
+                    methods = all ? DecompressionMethods.GZip : _all;
+                    break;
+
+#if !NETFRAMEWORK
+                case "br":
+                    if (IsWinHttpHandler)
+                    {
+                        // Brotli only supported on SocketsHttpHandler.
+                        return;
+                    }
+
+                    compress = s => new BrotliStream(s, CompressionLevel.Optimal, leaveOpen: true);
+                    methods = all ? DecompressionMethods.Brotli : _all;
+                    break;
+
+                case "deflate":
+                    // WinHttpHandler continues to use DeflateStream as it doesn't have a newer build than netstandard2.0
+                    // and doesn't have access to ZLibStream.
+                    compress = IsWinHttpHandler ?
+                        new Func<Stream, Stream>(s => new DeflateStream(s, CompressionLevel.Optimal, leaveOpen: true)) :
+                        new Func<Stream, Stream>(s => new ZLibStream(s, CompressionLevel.Optimal, leaveOpen: true));
+                    methods = all ? DecompressionMethods.Deflate : _all;
+                    break;
+#endif
+
+                default:
+                    Assert.Contains(encodingName, new[] { "br", "deflate", "gzip" });
+                    return;
             }
 
             var expectedContent = new byte[12345];
@@ -106,15 +113,15 @@ namespace System.Net.Http.Functional.Tests
         {
             yield return new object[]
             {
-                "deflate",
-                new Func<Stream, Stream>(s => new ZLibStream(s, CompressionLevel.Optimal, leaveOpen: true)),
+                "gzip",
+                new Func<Stream, Stream>(s => new GZipStream(s, CompressionLevel.Optimal, leaveOpen: true)),
                 DecompressionMethods.None
             };
 #if !NETFRAMEWORK
             yield return new object[]
             {
-                "gzip",
-                new Func<Stream, Stream>(s => new GZipStream(s, CompressionLevel.Optimal, leaveOpen: true)),
+                "deflate",
+                new Func<Stream, Stream>(s => new ZLibStream(s, CompressionLevel.Optimal, leaveOpen: true)),
                 DecompressionMethods.Brotli
             };
             yield return new object[]

--- a/src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs
+++ b/src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs
@@ -23,7 +23,7 @@ namespace System.IO.Compression
         /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream, compression mode, and whether to leave the <paramref name="stream"/> open.</summary>
         /// <param name="stream">The stream to which compressed data is written or from which decompressed data is read.</param>
         /// <param name="mode">One of the enumeration values that indicates whether to compress or decompress the stream.</param>
-        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
+        /// <param name="leaveOpen"><see langword="true" /> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <see langword="false" />.</param>
         public ZLibStream(Stream stream, CompressionMode mode, bool leaveOpen)
         {
             _deflateStream = new DeflateStream(stream, mode, leaveOpen, ZLibNative.ZLib_DefaultWindowBits);
@@ -39,7 +39,7 @@ namespace System.IO.Compression
         /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream, compression level, and whether to leave the <paramref name="stream"/> open.</summary>
         /// <param name="stream">The stream to which compressed data is written.</param>
         /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
-        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
+        /// <param name="leaveOpen"><see langword="true" /> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <see langword="false" />.</param>
         public ZLibStream(Stream stream, CompressionLevel compressionLevel, bool leaveOpen)
         {
             _deflateStream = new DeflateStream(stream, compressionLevel, leaveOpen, ZLibNative.ZLib_DefaultWindowBits);
@@ -87,7 +87,7 @@ namespace System.IO.Compression
         public override void SetLength(long value) => throw new NotSupportedException();
 
         /// <summary>Reads a byte from the stream and advances the position within the stream by one byte, or returns -1 if at the end of the stream.</summary>
-        /// <returns></returns>
+        /// <returns>The unsigned byte cast to an <see cref="int" />, or -1 if at the end of the stream.</returns>
         public override int ReadByte()
         {
             ThrowIfClosed();

--- a/src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs
+++ b/src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs
@@ -7,7 +7,7 @@ using System.Threading.Tasks;
 
 namespace System.IO.Compression
 {
-    /// <summary>Provides methods and properties used to compress and decompress streams by using the ZLib data format specification.</summary>
+    /// <summary>Provides methods and properties used to compress and decompress streams by using the zlib data format specification.</summary>
     public sealed class ZLibStream : Stream
     {
         /// <summary>The underlying deflate stream.</summary>
@@ -52,7 +52,7 @@ namespace System.IO.Compression
         public override bool CanWrite => _deflateStream?.CanWrite ?? false;
 
         /// <summary>Gets a value indicating whether the stream supports seeking.</summary>
-        public override bool CanSeek => _deflateStream?.CanSeek ?? false;
+        public override bool CanSeek => false;
 
         /// <summary>This property is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
         public override long Length => throw new NotSupportedException();

--- a/src/libraries/System.Net.Http.WinHttpHandler/tests/UnitTests/TestServer.cs
+++ b/src/libraries/System.Net.Http.WinHttpHandler/tests/UnitTests/TestServer.cs
@@ -160,7 +160,7 @@ namespace System.Net.Http.WinHttpHandlerUnitTests
             {
                 using (Stream compressedStream = useGZip ?
                     new GZipStream(memoryStream, CompressionMode.Compress) :
-                    new ZLibStream(memoryStream, CompressionMode.Compress))
+                    new DeflateStream(memoryStream, CompressionMode.Compress))
                 {
                     compressedStream.Write(bytes, 0, bytes.Length);
                 }

--- a/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs
+++ b/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs
@@ -211,6 +211,11 @@ namespace System.Net.Http
             { }
 
             protected override Stream GetDecompressedStream(Stream originalStream) =>
+                // As described in RFC 2616, the deflate content-coding is actually
+                // the "zlib" format (RFC 1950) in combination with the "deflate"
+                // compression algrithm (RFC 1951).  So while potentially
+                // counterintuitive based on naming, this needs to use ZLibStream
+                // rather than DeflateStream.
                 new ZLibStream(originalStream, CompressionMode.Decompress);
         }
 

